src/actions/sources/newSources.js

```diff
@@ -52,14 +52,12 @@ function loadSourceMaps(sources) {
       return;
     }
 
-    let originalSources = await Promise.all(
-      sources.map(({ id }) => dispatch(loadSourceMap(id)))
-    );
-
-    originalSources = flatten(originalSources).filter(Boolean);
-    if (originalSources.length > 0) {
-      await dispatch(newSources(originalSources));
-    }
+    sources.map(async ({ id }) => {
+      const originalSources = await dispatch(loadSourceMap(id));
```

Comment:
can we have always return an array? and enforce it with flow?

Reply:
Right, done.